### PR TITLE
Integrate MiniUPNP

### LINKED ISSUE #346: Support PCP for UPnP discovery and port forwarding
## Problem to solve

Our current implementation of UPnP is using UPnP v1/v2 and NAT PMP.
If it's working in some router, there is gap between miniupnpc and our implementation.
The gap would be the PCP UPnP support, which is the replacement of NAT PMP.

## Solution

We should support this implementation described here: http://miniupnp.free.fr/nat-pmp.html, to have better support UPnP networking discovery and port forwarding .

We should implement PCP to handle:

- [x] Integrate miniupnpc

### PR DESCRIPTION
**Pull request by samuel-uniris**
# Description

Integrate miniupnp as C dependency and build locally the binary to use it via system calls.
This way we are reproducing the same behavior.

This is quick fix to do not have to recreate in Erlang or Elixir the UPnP PCP implementation while keeping the support for NAT PMP and UPnP v1/v2

Fixes #346 

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested locally using `mix archethic.nettools` on Mac OS X (without upnpc installed) 

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,6 +8,9 @@ compile_c_programs:
 	mkdir -p priv/c_dist
 	$(CC) src/c/crypto/stdio_helpers.c src/c/crypto/ed25519.c -o priv/c_dist/libsodium_port -I src/c/crypto/stdio_helpers.h -lsodium
 	$(CC) src/c/hypergeometric_distribution.c -o priv/c_dist/hypergeometric_distribution -lgmp
+	
+	$(MAKE) -C src/c/nat/miniupnp/miniupnpc
```
Comment (apoorv-2204) [inline Makefile:12]:
```/usr/bin/gcc src/c/hypergeometric_distribution.c -o priv/c_dist/hypergeometric_distribution -lgmp
make -C src/c/nat/miniupnp/miniupnpc
make[1]: *** src/c/nat/miniupnp/miniupnpc: No such file or directory.  Stop.
make: *** [Makefile:12: compile_c_programs] Error 2```

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "src/c/nat/miniupnp"]
+	path = src/c/nat/miniupnp
+	url = https://github.com/miniupnp/miniupnp
```
Comment (apoorv-2204) [inline .gitmodules:3]: I think git failed to fetch, this lib

**Comment by apoorv-2204** [comment]
It is working locally for my pc.

- [ ] image
- [ ]

File: .gitmodules (hunk 2 above)
Comment (apoorv-2204) [reply .gitmodules:3]: - [ ] git submodule update --force --recursive --init --remote

Comment (samuel-uniris) [reply .gitmodules:3]: I think we can insert it into the Makefile

File: lib/archethic/networking/ip_lookup/nat_discovery/miniupnp.ex [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          |> to_charlist()
+          |> :inet.parse_address()
+          |> elem(1)
+
+        {_, status} ->
+          {:error, status}
+      end
+
+    opts = [
+      # Add redirection
```
Comment (apoorv-2204) [inline lib/archethic/networking/ip_lookup/nat_discovery/miniupnp.ex:42]: `-e "ArchethicPort"`
 Is it a good thing to have this?

Comment (samuel-uniris) [reply lib/archethic/networking/ip_lookup/nat_discovery/miniupnp.ex:42]: why not

Comment (apoorv-2204) [reply lib/archethic/networking/ip_lookup/nat_discovery/miniupnp.ex:42]: Do I do a that by committing on add_miniupnp?

Comment (samuel-uniris) [reply lib/archethic/networking/ip_lookup/nat_discovery/miniupnp.ex:42]: I will do it ;)

**Pull request merged**